### PR TITLE
[WIP] fix: Fields like name and email in custom forms should always be included

### LINKED ISSUE #6170: Incomplete data while fetching custom forms
Fields for name and email are missing while they are required, cannot be edited by organizer also.

Request:
image

Response:
```
{"data": [{"type": "custom-form", "attributes": {"is-included": true, "is-fixed": false, "type": "text", "is-required": true, "form": "attendee", "deleted-at": null, "field-identifier": "gender"}, "id": 20362, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20362/relationships/event", "related": "/v1/custom-forms/20362/event"}}}, "links": {"self": "/v1/custom-forms/20362"}}, {"type": "custom-form", "attributes": {"is-included": false, "is-fixed": false, "type": "text", "is-required": false, "form": "attendee", "deleted-at": null, "field-identifier": "address"}, "id": 20363, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20363/relationships/event", "related": "/v1/custom-forms/20363/event"}}}, "links": {"self": "/v1/custom-forms/20363"}}, {"type": "custom-form", "attributes": {"is-included": true, "is-fixed": false, "type": "text", "is-required": true, "form": "attendee", "deleted-at": null, "field-identifier": "city"}, "id": 20364, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20364/relationships/event", "related": "/v1/custom-forms/20364/event"}}}, "links": {"self": "/v1/custom-forms/20364"}}, {"type": "custom-form", "attributes": {"is-included": false, "is-fixed": false, "type": "text", "is-required": false, "form": "attendee", "deleted-at": null, "field-identifier": "country"}, "id": 20366, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20366/relationships/event", "related": "/v1/custom-forms/20366/event"}}}, "links": {"self": "/v1/custom-forms/20366"}}, {"type": "custom-form", "attributes": {"is-included": false, "is-fixed": false, "type": "text", "is-required": false, "form": "attendee", "deleted-at": null, "field-identifier": "jobTitle"}, "id": 20367, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20367/relationships/event", "related": "/v1/custom-forms/20367/event"}}}, "links": {"self": "/v1/custom-forms/20367"}}, {"type": "custom-form", "attributes": {"is-included": true, "is-fixed": false, "type": "text", "is-required": true, "form": "attendee", "deleted-at": null, "field-identifier": "phone"}, "id": 20368, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20368/relationships/event", "related": "/v1/custom-forms/20368/event"}}}, "links": {"self": "/v1/custom-forms/20368"}}, {"type": "custom-form", "attributes": {"is-included": false, "is-fixed": false, "type": "text", "is-required": false, "form": "attendee", "deleted-at": null, "field-identifier": "homeAddress"}, "id": 20371, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20371/relationships/event", "related": "/v1/custom-forms/20371/event"}}}, "links": {"self": "/v1/custom-forms/20371"}}, {"type": "custom-form", "attributes": {"is-included": false, "is-fixed": false, "type": "text", "is-required": false, "form": "attendee", "deleted-at": null, "field-identifier": "shippingAddress"}, "id": 20372, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20372/relationships/event", "related": "/v1/custom-forms/20372/event"}}}, "links": {"self": "/v1/custom-forms/20372"}}, {"type": "custom-form", "attributes": {"is-included": true, "is-fixed": false, "type": "text", "is-required": false, "form": "attendee", "deleted-at": null, "field-identifier": "company"}, "id": 20373, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20373/relationships/event", "related": "/v1/custom-forms/20373/event"}}}, "links": {"self": "/v1/custom-forms/20373"}}, {"type": "custom-form", "attributes": {"is-included": false, "is-fixed": false, "type": "text", "is-required": false, "form": "attendee", "deleted-at": null, "field-identifier": "website"}, "id": 20376, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20376/relationships/event", "related": "/v1/custom-forms/20376/event"}}}, "links": {"self": "/v1/custom-forms/20376"}}, {"type": "custom-form", "attributes": {"is-included": false, "is-fixed": false, "type": "text", "is-required": false, "form": "attendee", "deleted-at": null, "field-identifier": "blog"}, "id": 20377, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20377/relationships/event", "related": "/v1/custom-forms/20377/event"}}}, "links": {"self": "/v1/custom-forms/20377"}}, {"type": "custom-form", "attributes": {"is-included": true, "is-fixed": true, "type": "text", "is-required": true, "form": "session", "deleted-at": null, "field-identifier": "title"}, "id": 20381, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20381/relationships/event", "related": "/v1/custom-forms/20381/event"}}}, "links": {"self": "/v1/custom-forms/20381"}}, {"type": "custom-form", "attributes": {"is-included": false, "is-fixed": false, "type": "select", "is-required": false, "form": "session", "deleted-at": null, "field-identifier": "language"}, "id": 20382, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20382/relationships/event", "related": "/v1/custom-forms/20382/event"}}}, "links": {"self": "/v1/custom-forms/20382"}}, {"type": "custom-form", "attributes": {"is-included": true, "is-fixed": false, "type": "text", "is-required": false, "form": "session", "deleted-at": null, "field-identifier": "shortAbstract"}, "id": 20384, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20384/relationships/event", "related": "/v1/custom-forms/20384/event"}}}, "links": {"self": "/v1/custom-forms/20384"}}, {"type": "custom-form", "attributes": {"is-included": false, "is-fixed": false, "type": "text", "is-required": false, "form": "session", "deleted-at": null, "field-identifier": "subtitle"}, "id": 20386, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20386/relationships/event", "related": "/v1/custom-forms/20386/event"}}}, "links": {"self": "/v1/custom-forms/20386"}}, {"type": "custom-form", "attributes": {"is-included": false, "is-fixed": false, "type": "text", "is-required": false, "form": "session", "deleted-at": null, "field-identifier": "longAbstract"}, "id": 20388, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20388/relationships/event", "related": "/v1/custom-forms/20388/event"}}}, "links": {"self": "/v1/custom-forms/20388"}}, {"type": "custom-form", "attributes": {"is-included": true, "is-fixed": true, "type": "text", "is-required": true, "form": "speaker", "deleted-at": null, "field-identifier": "name"}, "id": 20390, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20390/relationships/event", "related": "/v1/custom-forms/20390/event"}}}, "links": {"self": "/v1/custom-forms/20390"}}, {"type": "custom-form", "attributes": {"is-included": false, "is-fixed": false, "type": "text", "is-required": false, "form": "speaker", "deleted-at": null, "field-identifier": "github"}, "id": 20392, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20392/relationships/event", "related": "/v1/custom-forms/20392/event"}}}, "links": {"self": "/v1/custom-forms/20392"}}, {"type": "custom-form", "attributes": {"is-included": false, "is-fixed": false, "type": "text", "is-required": false, "form": "attendee", "deleted-at": null, "field-identifier": "state"}, "id": 20365, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20365/relationships/event", "related": "/v1/custom-forms/20365/event"}}}, "links": {"self": "/v1/custom-forms/20365"}}, {"type": "custom-form", "attributes": {"is-included": false, "is-fixed": false, "type": "text", "is-required": false, "form": "attendee", "deleted-at": null, "field-identifier": "taxBusinessInfo"}, "id": 20369, "relationships": {"event": {"links": {"self": "/v1/custom-forms/20369/relationships/event", "related": "/v1/custom-forms/20369/event"}}}, "links": {"self": "/v1/custom-forms/20369"}}], "links": {"self": "/v1/events/910/custom-forms", "first": "/v1/events/910/custom-forms", "last": "/v1/events/910/custom-forms?page%5Bnumber%5D=3", "next": "/v1/events/910/custom-forms?page%5Bnumber%5D=2"}, "meta": {"count": 55}, "jsonapi": {"version": "1.0"}}
```

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #6170 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

- [X] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by prateekj117** [comment]
@mrsaicharan1 @iamareebjamal Please review.

**Comment by prateekj117** [comment]
@uds5501 Yes, I think that can be done as well. We can see for all event_id if the three fields are present in their custom form. And create the ones if they are not present. @iamareebjamal Do we need to handle that case as well?

**Comment by iamareebjamal** [comment]
Apply same logic when you get custom form

**Comment by prateekj117** [comment]
@iamareebjamal In the `before_get_object`, we are checking for event_id, while we have no Url which is trying to get custom form using the event_id. Just one plain simple `custom-forms/<int:id>`. I think that logic is redundant and should be removed.

**Comment by iamareebjamal** [comment]
That would be another issue unrelated to this

File: app/api/custom_forms.py [hunk 1]
```diff
@@ -74,6 +84,18 @@ def before_get_object(self, view_kwargs):
...
+        if custom_form:
+            if custom_form.form == 'attendee':
+                event_id = custom_form.event_id
+                get_or_create(CustomForms, field_identifier='firstname', form='attendee', type='text', is_required=True,
+                              is_included=True, is_fixed=True, event_id=event_id)
+                get_or_create(CustomForms, field_identifier='lastname', form='attendee', type='text', is_required=True,
+                              is_included=True, is_fixed=True, event_id=event_id)
+                get_or_create(CustomForms, field_identifier='email', form='attendee', type='email', is_required=True,
+                              is_included=True, is_fixed=True, event_id=event_id)
+                db.session.commit()
```
Comment (iamareebjamal) [inline app/api/custom_forms.py]: Duplicate code

File: app/api/custom_forms.py [hunk 2]
```diff
@@ -13,6 +13,17 @@
 from app.models.event import Event
 
 
+def required_fields(form, event_id):
```
Comment (iamareebjamal) [inline app/api/custom_forms.py]: get_or_create_required_fields

Comment (prateekj117) [reply app/api/custom_forms.py]: @iamareebjamal Done

**Comment by prateekj117** [comment]
@iamareebjamal Ready for Review.

**Comment by prateekj117** [comment]
@CosmicCoder96 Oh, I missed that. @iamareebjamal If this was already there, how was https://github.com/fossasia/open-event-server/issues/6170#issuecomment-512756909 missed?

**Comment by iamareebjamal** [comment]
Don't know. Try to reproduce the bug

**Comment by prateekj117** [comment]
@iamareebjamal I think we should remove the function from after event creation hook. What if the event owner doesn't want to use an attendee custom form?

**Comment by iamareebjamal** [comment]
No

**Comment by prateekj117** [comment]
In that case, I will make changes accordingly. Though, can you help me understand why?

**Comment by iamareebjamal** [comment]
We want things to be consistent. If owner does not want to use the form, it'll be ignored

**Comment by prateekj117** [comment]
@iamareebjamal Closing this issue. Can't think of any reason why the error was occurring in that particular custom-form id.

**Pull request closed without merging**